inc/namespace.php

```diff
@@ -5,14 +5,40 @@
 use Exception;
 use function HM\Platform\get_aws_sdk;
 
-/**
- * Bootstrapper for the plugin.
+/*
+ * Set initial values and register handlers
+ */
+function initialize() {
+	$GLOBALS['hm_platform_xray_errors'] = [];
+
+	global $hm_platform_xray_start_time;
+	if ( ! $hm_platform_xray_start_time ) {
+		$hm_platform_xray_start_time = microtime( true );
+	}
+
+	if ( ! defined( 'AWS_XRAY_DAEMON_IP_ADDRESS' ) ) {
+		define( 'AWS_XRAY_DAEMON_IP_ADDRESS', '127.0.0.1' );
+	}
+
+	ini_set( 'xhprof.sampling_interval', 5000 ); // @codingStandardsIgnoreLine
+	xhprof_sample_enable();
+
+	register_handlers();
+}
+
+/*
+ * Register error handler and shutdown function
  */
-function bootstrap() {
-	add_action( 'shutdown', __NAMESPACE__ . '\\on_shutdown', 99 );
-	add_filter( 'query', __NAMESPACE__ . '\\filter_mysql_query' );
-	add_action( 'requests-requests.before_request', __NAMESPACE__ . '\\trace_requests_request', 10, 5 );
-	set_error_handler( __NAMESPACE__ . '\\error_handler', error_reporting() ); // @codingStandardsIgnoreLine
+function register_handlers() {
+	register_shutdown_function( __NAMESPACE__ . '\\on_shutdown' );
+
+	$current_errror_handler = set_error_handler( function () use ( &$current_errror_handler ) { // @codingStandardsIgnoreLine
+		call_user_func_array( __NAMESPACE__ . '\\error_handler', func_get_args() );
+		if ( $current_errror_handler ) {
+			call_user_func_array( $current_errror_handler, func_get_args() );
+		}
+	} );
+
 	send_trace_to_daemon( get_in_progress_trace() );
```

Comment:
This function isn't really `wp-cache-bucket` as it calls `send_trace_to_daemon

Reply:
Would you clarify this comment further? I'm not sure where `wp-cache-bucket` comes into play 🤔

Reply:
Haha sorry I had somethign else in my clipboard! It looks like you renamed the function to `register_handlers`, but the function does more than register handlers, it also kicks off a call to `send_trace_to_daemon`, so I think this is now badly named.

Reply:
Ah, makes sense. Looking at this now, I wonder if I should just include all this in `initialize()`. All this is only supposed to be run once, so no need for a second function.

Reply:
Okie doke, I went ahead and did that. I also renamed `initialize()` to `bootstrap()` for consistency